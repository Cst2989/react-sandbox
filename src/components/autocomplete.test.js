import React from 'react';
import ReactDOM from "react-dom";
import { shallow } from 'enzyme';
import Autocomplete from './Autocomplete';
import { create, act } from "react-test-renderer";
const MockedData = [
    {
        name: 'Dan Neciu',
        initials: 'DN',
        avatar: null,
        email: 'dan.neciu@fakee.com'
    },
    {
        name: 'Joe Johnson',
        initials: 'JJ',
        avatar: null,
        email: 'joe@fakee.com'
    },
    {
        name: 'Ion Ionescu',
        initials: 'IJ',
        avatar: null,
        email: 'fake@fakee.com'
    },
    {
        name: 'Donald Trump',
        initials: 'DT',
        avatar: null,
        email: 'donald@fakee.com'
    },
    {
        name: 'JR Smith',
        initials: 'JR',
        avatar: null,
        email: 'jerry@fakee.com'
    }
];
const createNodeMock = element => {
    if (element.type === 'li') {
        // This is your fake DOM node for <p>.
        // Feel free to add any stub methods, e.g. focus() or any
        // other methods necessary to prevent crashes in your components.
        return {
            scrollIntoView : function (){
                
            }
        };
    }
    // You can return any object from this method for any type of DOM component.
    // React will use it as a ref instead of a DOM node when snapshot testing.
    return null;
}
it('should have a input regardless of props', () => {
    const autocomplete = shallow(<Autocomplete />);
    const input = autocomplete.find('input').last();
    expect(input.length).toEqual(1);
});

it('should render a list based on the mocked data', () => {
    const options = {createNodeMock};
    let autocomplete; 
    act(() => {
        autocomplete = create(<Autocomplete data={ MockedData} />, options);
    });
    const testInstance = autocomplete.root;
    const input = testInstance.findAll(
        (el) => el.type === 'li'
    );
    expect(input.length).toEqual(5);
});


