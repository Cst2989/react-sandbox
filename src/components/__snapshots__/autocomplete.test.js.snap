// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> with mocked data should match the snapshot 1`] = `"<div class=\\"undefined \\"><label class=\\"undefined\\">Managers</label><input type=\\"text\\" data-active=\\"false\\" data-active-item=\\"1\\" class=\\"undefined\\" placeholder=\\"Search for Managers...\\" value=\\"\\"><div class=\\"undefined\\"></div><ul class=\\"undefined\\"><li class=\\"undefined undefined\\"><span class=\\"undefined\\">DN</span><span class=\\"undefined\\"><span class=\\"undefined\\">Dan Neciu</span><span class=\\"undefined\\">dan.neciu@fakee.com</span></span></li><li class=\\"undefined \\"><span class=\\"undefined\\">JJ</span><span class=\\"undefined\\"><span class=\\"undefined\\">Jon Nehnson</span><span class=\\"undefined\\">joe@fakee.com</span></span></li><li class=\\"undefined \\"><span class=\\"undefined\\">IJ</span><span class=\\"undefined\\"><span class=\\"undefined\\">Ion Ionescu</span><span class=\\"undefined\\">fake@fakee.com</span></span></li><li class=\\"undefined \\"><span class=\\"undefined\\">DT</span><span class=\\"undefined\\"><span class=\\"undefined\\">Donald Trump</span><span class=\\"undefined\\">donald@fakee.com</span></span></li><li class=\\"undefined \\"><span class=\\"undefined\\">JR</span><span class=\\"undefined\\"><span class=\\"undefined\\">JR Smith</span><span class=\\"undefined\\">jerry@fakee.com</span></span></li></ul></div>"`;

exports[`<Autocomplete /> with no props should match the snapshot 1`] = `"<div class=\\"undefined \\"><label class=\\"undefined\\">Default Label</label><input type=\\"text\\" data-active=\\"false\\" data-active-item=\\"1\\" class=\\"undefined\\" value=\\"\\" placeholder=\\"Default Placeholder\\"/><div class=\\"undefined\\"></div></div>"`;
